pandas-stubs/core/generic.pyi

```diff
@@ -432,6 +439,21 @@ class NDFrame(indexing.IndexingMixin):
         end_time,
         axis=...,
     ) -> Self: ...
+    @final
+    def resample(
+        self,
+        rule: Frequency,
+        axis: Axis | NoDefault = ...,
+        closed: Literal["right", "left"] | None = None,
```

Comment:
So far, we have tried to not have the default values in the stubs. (I believe pyright/mypy(?) have an option to create stubs with/without default values as people requested that having default values in stubs can be helpful.)

Reply:
The same thing was done at typeshed until a year ago where the [view has shifted towards keeping the default values](https://github.com/python/typeshed/issues/8988). Since then, types checkers and linters like flake8-pyi have been updated to not flag simple default values in stubs and stubtest was updated to include a check for default values so they do not go out of sync. I also recently [modified stubgen](https://github.com/python/mypy/commit/e69c5cde8643e04a54a644cc27814ab98181541d) to preserve default values in generated stubs unconditionally; this change was included in the latest release of mypy.

Default values, although not very useful for command line type checkers, proved useful for other applications like IDEs that use the stubs like VSCode and Pycharm and thus are now encouraged.